lib/stdlib/test/property_test/queue_prop.erl

```diff
@@ -0,0 +1,670 @@
+				{L1 =:= [], Q1, L1};
+			    {{value, I}, Q2} when Op =:= out ->
+				{I =:= hd(L1), Q2, tl(L1)};
+			    {{value, I}, Q2} when Op =:= out_r ->
+				{I =:= lists:last(L1), Q2, lists:droplast(L1)}
+			end;
+		    % queue:peek/1, queue:peek_r/1
+		    (Op, {true, Q1, L1}) when Op =:= peek; Op =:= peek_r ->
+			case queue:Op(Q1) of
+			    empty ->
+				{L1 =:= [], Q1, L1};
+			    {value, I} when Op =:= peek ->
+				{I =:= hd(L1), Q1, L1};
+			    {value, I} when Op =:= peek_r ->
+				{I =:= lists:last(L1), Q1, L1}
+			end;
+		    % queue:get/1, queue:head/1
+		    (Op, {true, Q1, L1}) when Op =:= get; Op =:= head ->
+			try queue:Op(Q1) of
+			    I ->
+				{I =:= hd(L1), Q1, L1}
+			catch error:empty ->
+			    {L1 =:= [], Q1, L1}
+			end;
+		    % queue:get_r/1, queue:daeh/1, queue:last/1
+		    (Op, {true, Q1, L1}) when Op =:= get_r; Op =:= daeh; Op =:= last ->
+			try queue:Op(Q1) of
+			    I ->
+				{I =:= lists:last(L1), Q1, L1}
+			catch error:empty ->
+			    {L1 =:= [], Q1, L1}
+			end;
+		    % queue:init/1, queue:drop_r/1, queue:liat/1
+		    (Op, {true, Q1, L1}) when Op =:= init; Op =:= drop_r; Op =:= liat ->
+			try queue:Op(Q1) of
+			    Q2 ->
+				L2 = lists:droplast(L1),
+				{equal(L2, Q2), Q2, L2}
+			catch error:empty ->
+			    {L1 =:= [], Q1, L1}
+			end;
+		    % queue:drop/1, queue:tail/1
+		    (Op, {true, Q1, L1}) when Op =:= drop; Op =:= tail ->
+			try queue:Op(Q1) of
+			    Q2 ->
+				L2 = tl(L1),
+				{equal(L2, Q2), Q2, L2}
+			catch error:empty ->
+			    {L1 =:= [], Q1, L1}
+			end
+		end,
+		{true, Q, L},
+		Ops
+	    ),
+	    Res andalso equal(RL, RQ)
+	end
+    ).
+
+%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Property helpers %%%
+%%%%%%%%%%%%%%%%%%%%%%%%
+
+% get/1 and head/1 have the same semantics
+common_get_head(Fn) ->
+    ?FORALL(
+	{L, Q},
+	list_queue(),
+	try
+	    queue:Fn(Q)
+	of E ->
+	    E =:= hd(L)
+	catch error:empty ->
+	    L =:= []
+	end
+    ).
+
+% get_r/1, last/1 and daeh/1 have the same semantics
+common_get_r_last_daeh(Fn) ->
+    ?FORALL(
+	{L, Q},
+	list_queue(),
+	try
+	    queue:Fn(Q)
+	of E ->
+	    E =:= lists:last(L)
+	catch error:empty ->
+	    L =:= []
+	end
+    ).
+
+% drop_r/1, init/1 and liat/1 have the same semantics 
+common_drop_r_init_liat(Fn) ->
+    ?FORALL(
+	{L, Q},
+	list_queue(),
+	try
+	    queue:Fn(Q)
+	of Q1 ->
+	    equal(lists:droplast(L), Q1)
+	catch error:empty ->
+	    L =:= []
+	end
+    ).
+
+% drop/1 and tail/1 have the same semantics
+common_drop_tail(Fn) ->
+    ?FORALL(
+	{L, Q},
+	list_queue(),
+	try
+	    queue:Fn(Q)
+	of Q1 ->
+	    equal(tl(L), Q1)
+	catch error:empty ->
+	    L =:= []
+	end
+    ).
+
+% in_r/2 and cons/2 have the same semantics
+common_in_r_cons(Fn) ->
+    ?FORALL(
+	L,
+	list(),
+	begin
+	    Q = lists:foldl(
+		fun (I, Acc) ->
+		    queue:Fn(I, Acc)
+		end,
+		queue:new(),
+		L
+	    ),
+	    equal(lists:reverse(L), Q)
+	end
+    ).
+
+%%%%%%%%%%%%%%%%%%
+%%% Generators %%%
+%%%%%%%%%%%%%%%%%%
+
+list_queue() ->
```

Comment:
See comment on `prop_join` for a suggestion on how to improve this generator.

Reply:
See the comment on your comment :grin: